### PR TITLE
update AmericanOption notebook; fix docstring in fd_solvers

### PR DESCRIPTION
**Pull request by kykosic**
Changes:
- Updated the American Options notebook to use the current v14 API and TF 2.1
- Fixed a docstring in fd_solvers for the correct required callable signature.

Resolves issue #17

### CODE REVIEW

File: tf_quant_finance/examples/jupyter_notebooks/American_Option_Black_Scholes.ipynb [hunk 1]
```diff
@@ -1,958 +1,950 @@
...
...CEqSJEnfSmarg1c/LeZKg6gBGzU8lP/3qUlYzXa39xxt6mJvvVg+9tFp\neSItluGBt24kUVWVkpIvKC4+D4DR6ENe3iLCw933Jb4dTqfCgc/Kqa0WXSvCo/zJX5WNr593d6Yq\nqsKGyi2caDwrxmUKZW3us0T4um/FN5TMXeW01/7/7L1neBRXtqj9VmflnEWQEJJAIBBgcjTJGEww\n2MYYnMEJh/HYM2fu9+vc57lzz5zxJOc0DuMENsEmmpyNCQYFhBBISEgop1bq3LW/H8XROXORPAa3\naWHv9x90dbGqulG9WnuvtTYAKqAQ1W8RQZFa257K87l8u+1DVNWLougYPfs++mWOwHHgPTznDwOg\nBEdhHPYM9hNWECCESlvx1zgbiom9bwXh06b77+L8hBRBiUQikdx0tNtc/HltHpfqtHFg2QOieHLh\nECwmQ7ciKIRgb3Uze6q15eMAvY6HM5JICrp6CVYIwalTxykszAPAYglg5sy5RET4ppmz2+3l6/WF\nXC7XWp3EJoQw756hmC09L2XfCLyql4+KPudEnTZ/NzYwmmeGryLC4v/+hQAdTbk0V2wGBCh6olOW\nEBiWAUDZ2eOc2PHJlUIePePmPkhSahaOPW/gKTsJgC4sHn3mk9hPavddqB5ai7bgaikn/uGVhI6f\n4K9L8ytSBCUSiURyU9Ha4eSltblUNWiVoiPTY3hsQRYGffcb+4UQfH25iUO1mgAEG/Q8nJFEfODV\nS7BCCE6c+IZz57Q5tIGBQcycOZewMN/IkMvpYdu6M9RUtgKQkBzG7XcNwWT27+PYo3p4v/AzchsK\nAEgMimf18JWEmf3fugagveE4LZe/BkDRGYlJvQdLSCoAF3IPcWrvFwDoDUYmzH+U+ORU7Dv/hrdS\nux4lsi+61Edx5GrfAdXrorXwK9wdNSQ8/iQhI2/xw1X1DqQISiQSieSmobnNwR/X5FLXrFWDjs2K\n45G5g9D3UN2pCsGWiga+rdfEK8xo4JHMJKItVy/BqqrKt98eoqSkGIDg4BBmzpxLSIhvmjk7HW62\nrC2gvkbLYib3j+C2xVkYjXqfnP96cXndvHvmIwqbzgHQJySJ1cMfJdjo/9Y1AK21h2it0YpWFL2Z\n2NRlmIO1yuWi47vIP7wZAKPJwqSFjxEdm4h925/w1mrj+ZTYNH
```
Comment (matejr) [inline tf_quant_finance/examples/jupyter_notebooks/American_Option_Black_Scholes.ipynb]: Are you using our pip package? Can you fix this?

Comment (cyrilchim) [inline tf_quant_finance/examples/jupyter_notebooks/American_Option_Black_Scholes.ipynb]: Please remove Warning line.

Comment (kykosic) [reply tf_quant_finance/examples/jupyter_notebooks/American_Option_Black_Scholes.ipynb]: I added a fix to find the sobol data when installed in editable mode via `pip install -e .`.

Comment (kykosic) [reply tf_quant_finance/examples/jupyter_notebooks/American_Option_Black_Scholes.ipynb]: I solved the warning for local installs.

**Comment by kykosic** [comment]
I've edited the notebook to use 500 options by default for CPU benchmark. The results look a lot better now.

**Comment by cyrilchim** [comment]
Thanks Kyle! It's all submitted now.

**Comment by kykosic** [comment]
Awesome thanks!

**Pull request merged**